### PR TITLE
288 allow ellipsis in `get_code`

### LINKED ISSUE #288: Allow `teal.data::get_code.teal_data` to use `...`
So that you can pass `check_names` in `teal.data::get_code.teal_data` to `teal.data:::get_code_dependency`.

There is only one place where we actually use `teal.data:::get_code_dependency` and it's here https://github.com/insightsengineering/teal/pull/1021#discussion_r1479451436
We would be able to actually use `teal.data::get_code.teal_data()` which is exported if we all this method to use `...`.

### PR DESCRIPTION
**Pull request by m7pr**
Close #288 

Allows to pass more parameters in `get_code` through `...`. For example you can now pass `check_names` 

```r
code <- c(
  "a <- 1",
  "b <- identity(x = a)",
  "a <- 2"
)
tdata <- eval_code(teal_data(), code)
get_code(tdata, datanames = "c", check_names = TRUE)
#> character(0)
Warning message:
In get_code_dependency(object@code, datanames, ...) :
  Object(s) not found in code: c
  

get_code(tdata, datanames = "c", check_names = FALSE)
#> character(0)
```

### CODE REVIEW

File: R/teal_data-get_code.R
Comment (m7pr) [reply R/teal_data-get_code.R:80]: Yeah, I was thinking we could put something like `[teal.data:::get_code_dependency()]` but was unsure you can actually have it in the documentation with `:::` and whether exposition of internal function in the documentation is a good idea.
What if we change to 

```r
Parameters passed to internal methods. Only for advanced usage.
```

This `only for advanced usage` phrase is common in a `tidyverse` ecosystem.

Comment (chlebowa) [reply R/teal_data-get_code.R:80]: Agreed. Enumerate the allowed arguments explicitly, possibly in `Details`.

Comment (m7pr) [reply R/teal_data-get_code.R:80]: # Proposition 1

```r
@param ... Parameters passed to internal methods. Currently, the only supported parameter is `check_names` (`logical(1)`) flag, which is `TRUE` by default. Function warns about missing objects, if they do not exist in `code` but are passed in `names`. To remove the warning, set `check_names = FALSE`.
```

# Proposition 2

```r
@param ... Parameters passed to internal methods. See `Details`.
```
and in `Details` section.

```r
For `...`, currently, the only supported parameter is `check_names` (`logical(1)`) flag, which is `TRUE` by default. Function warns about missing objects, if they do not exist in `code` but are passed in `names`. To remove the warning, set `check_names = FALSE`.
```

**Comment by pawelru** [comment]
The change is significant. Please add NEWS enty

File: R/teal_data-get_code.R
Comment (averissimo) [reply R/teal_data-get_code.R]: Looks good!

File: NEWS.md
Comment (averissimo) [inline NEWS.md:10]: It's actually under `teal.data 0.4.0`, please move up and add the `### Enhancements` section :slightly_smiling_face:

**Review by averissimo — APPROVED** [review]
The change looks good, sorry for not voting on the proposal, I missed the notification.

One minor change as the `NEWS.md` entry seems to be under the already released `0.4.0` version.

Don't merge this as it needs to go in after #294 _(unless @donyunardi wants to add it to the patch release)_

File: NEWS.md
Comment (m7pr) [reply NEWS.md:10]: Yes, totally - also mentioned this here https://github.com/insightsengineering/teal.data/pull/290#issuecomment-1933619133

Comment (m7pr) [reply NEWS.md]: moved in here 8fec7927f8c2ad3856308474e751a16fe60d5e4e

**Pull request merged**
